store/sqlstore/channel_store.go

```diff
@@ -3183,3 +3423,569 @@ func (s SqlChannelStore) GroupSyncedChannelCount() (int64, *model.AppError) {
 
 	return count, nil
 }
+
+type sidebarCategoryForJoin struct {
+	model.SidebarCategory
+	ChannelId *string
+}
+
+func (s SqlChannelStore) CreateSidebarCategory(userId, teamId string, newCategory *model.SidebarCategoryWithChannels) (*model.SidebarCategoryWithChannels, *model.AppError) {
```

Comment:
nit: We should return `error` with the new error approach for the store. Not an stopper here, but something to keep in mind for the future (This is going to need migration in the future).

Reply:
This method was actually added before we started the `error` migration, so it needed it regardless 😛 

I started that migration on a few of the other store methods that I touched more recently, but this wasn't one of them. I'll keep it in mind since there's still a couple outstanding issues that might be touching this code in the next week or two.